assignment2.c

```diff
@@ -0,0 +1,187 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+
+#define FILE_NAME "users.txt"
+
+typedef struct {
+    int id;
+    char name[25];
+    int age;
+} User;
+
+
+// function to create a file if it doesn't already exists
+void createFile(){
+    FILE *fptr = fopen(FILE_NAME, "a");
+    if(fptr) fclose(fptr); 
+}
+
+
+// function to add user to file
+void addUser(){
+    FILE *fptr = fopen(FILE_NAME, "a");
+
+    if(!fptr){
+        printf("Error opening file \n");
+        return;
+    }
+
+    User user;
+    printf("enter user id : ");
+    scanf("%d", &user.id);
+    getchar();
+    printf("enter name of user : ");
+    fgets(user.name, sizeof(user.name), stdin);
+    user.name[strcspn(user.name, "\n")] = 0;
+    printf("enter age of user : ");
+    scanf("%d", &user.age);
+    
+    fprintf(fptr, "%d, %s, %d\n", user.id, user.name, user.age);
+    fclose(fptr);
+    printf("User added successfull \n");
+}
+
+
+// function to display existing users
+void displayUsers(){
+    FILE *fptr = fopen(FILE_NAME, "r");
+    if(!fptr){
+        printf("Error opening file");
+        return;
+    }
+
+    User user;
+    while(fscanf(fptr, "%d,%49[^,],%d\n", &user.id, user.name, &user.age) != EOF){
+        printf("%d\t%-25s%d\n", user.id, user.name, user.age);
+    }
+
+    fclose(fptr);
+}
+
+
+// function to update user data
+void updateUser(){
+    FILE *fptr = fopen(FILE_NAME, "r");
+    if(!fptr){
+        printf("Error opening file");
+        return;
+    }
+
+    int id, found = 0;
+    printf("enter id of user to update: ");
+    scanf("%d", &id);
+    getchar();
+
+    User user;
+    FILE *temp = fopen("temp.txt", "w");
```

Comment:
Temp file is created here because whenever we open a file in write mode, all the existing contents are erased and hence the updation could not be done using only original file as all existing data would be lost. And append mode cannot be used for the purpose.
Here we have copied entire records in a temp file along with the updation of required one and later replaced temp fiile with the original file.